src/libraries/Common/src/Interop/Unix/System.Native/Interop.Stat.Span.cs

```diff
@@ -9,16 +9,12 @@ internal static partial class Interop
 {
     internal static partial class Sys
     {
-        // Unix max paths are typically 1K or 4K UTF-8 bytes, 256 should handle the majority of paths
-        // without putting too much pressure on the stack.
-        private const int StackBufferSize = 256;
-
         [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_Stat", SetLastError = true)]
         internal static extern int Stat(ref byte path, out FileStatus output);
 
         internal static int Stat(ReadOnlySpan<char> path, out FileStatus output)
         {
-            var converter = new ValueUtf8Converter(stackalloc byte[StackBufferSize]);
+            var converter = new ValueUtf8Converter(stackalloc byte[DefaultPathBufferSize]);
```

Comment:
question: can this be `using var` instead of calling `Dispose`?

Reply:
No, `ValueUtf8Converter` is not disposable.

@GrabYourPitchforks Do you know why it was decided _not_ to make this type disposable? Is there anything special about its `Dispose` method that would make it unsafe to call inside a `using`?

https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/src/libraries/Common/src/System/Text/ValueUtf8Converter.cs#L13

Reply:
If you do `IDisposable` you'd need boxing if anyone ever tries to store it or pass it as `IDisposable`; also, ref structs can't implement interfaces (precisely because they can't be boxed). You can still do `using var` even if it doesn't implement `IDisposable`: [SharpLab](https://sharplab.io/#v2:D4AQDABCCMB0BCBXAZsgpgJwM4G4CwAUIQHYCGAtmlgA6kDGaUcAKmgB4AuhA3oRPxAD0wiAB4sicuVIYAngD4+A4YIgAJNABtqmCBwD2EUps36A7hEQcAlpusAva8QDmELB3oBrCACMU6DAhkfUC6fWIAN0wsa3C9QwBVZgAxAFoADlgIAHVbTSV+FTczaw46AAt4iABBDAxSWQAFfX1NCGtkCGJ9DghnayjiCDRuxGdKynIQ2SzqrDdECoAaCGlPRgkMRgMCoRE6YzaAYU00GQAKAEoqkAB2I2JZCGoekZtDp63iDjQAE19/LpSMgfqFwlFsLFiLBdkVRIIJFIZApdk5QWQ2ltOu4MItegA1YyINAJDjIdJHcGYUEQABcEAAkgARaw0fRYUg+U67XgEAT854YAakH6+WQ/ADaAF0APwQAD6MnqsmY+gASmgOIgMMRVc1Wvg+QL+NQhRERYwAMq0YiiHzitDyBX2n64Qi7fkgADMEEJmmJpPJlMi1Mw52tpFtLsd7WIpWsxiQqEwlw9Al5xuNirqDVVGq1Or1LTaAF4uogTIbMwL5dH5mWnPHE4CMFXMwBfd1G43eiARqMOp3BiEcarEX6sDDkJwWy0cIUuc4a0i/ADyxE0sn7ogqMid5v9aFT3YFGerAjRq1IbEtDkYZYAosQwr8nM5YElkpkAOKagCy17wA6lKIN85wHsSsAADIjM4HDlNcADUEDQG257tJ05y1g6WDQbB8FiFeN53se6H8GeZECCybJYGgVxoZRCpKrm6qatqur6PqpY1DmTTFnag6wJa5QyH8sAamB0jEfYR4MZR2GuhAZbEGgFjbtG8hYcxKqsQWHFcZcckdl2lEiN+9Q+Ho5SMHWDz/MQFZtKC05kD8abGpe0Ygd8SkQE+L5vh+KQ/pqQGuuBRJoCsClUIZ7k1nWEpefooEcFKvlgEZPb3DFuGWnYDDnGAKzJalEDIdAcUnvynZENVAi9iAAAsEDUS8tFXPFFHntG0pygY+bsb52bKnmbGFpxxZZQKHQQOcA3jUMACEymOaRZHdfJ2ljXpRatL5DmVvF1a1MqXECT88hCSJWy/OJi3zbp7FVZRtXVm9AgffwuyNQATEwtw8vFjXQAAbLGvQAU4nX1eRx0NZA5qBOwB2qb6kWBhSVIYKC5zuF4hz6HQEBhWgErQGAYBSi9ZF3BAmXxV9EC1e2QA===)

Reply:
Interesting, thanks for showing me @hoyosjs . Didn't know `using` could be used with non-`IDisposable`s.

We have a list of to-dos for changes that are unrelated to this PR. We'll add this suggestion to the list.